homeassistant/components/local_ip/config_flow.py

```diff
@@ -14,18 +15,16 @@ class SimpleConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
-        if user_input is not None:
-            if any(
-                user_input["name"] == entry.data["name"]
-                for entry in self._async_current_entries()
-            ):
-                return self.async_abort(reason="already_configured")
 
-            return self.async_create_entry(title=user_input["name"], data=user_input)
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is not None:
+            return self.async_create_entry(title=user_input[CONF_NAME], data=user_input)
 
         return self.async_show_form(
             step_id="user",
-            data_schema=vol.Schema({vol.Required("name", default=DOMAIN): str}),
+            data_schema=vol.Schema({vol.Required(CONF_NAME, default=DOMAIN): str}),
```

Comment:
If there's only a single instance allowed, do we still need the name option?

Reply:
Just for the user to display the name he wants ?

But he still can change the name of the sensor via the entity settings in the UI ...

Should I remove it ?

Reply:
Yeah,I think we can remove it here, since the user has this option via entity registry.